### ISSUE TITLE
gnome-software crash

### ISSUE THREAD

**Issue by davidak** [NixOS/nixpkgs#40253]
## Issue description

even if if would work, what is the benefit of having it (in default gnome desktop)? maybe just remove it.

cc @jtojnar 

### Steps to reproduce

1. use gnome desktop
2. `gnome-software --verbose`

https://pastebin.com/TQcbXQ30
```
...
21:14:43:0364 Gs  Failed to map ISO639 to language names: cannot find source file : '/usr/pkg/share/xml/iso-codes/iso_639.xml'
Trace/breakpoint trap
```

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.39, NixOS, 18.03.132275.2352d46904d (Impala)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0.1`
 - channels(root): `"nixos-18.03.132275.2352d46904d"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by jtojnar** [comment]
This was fixed as a part of https://github.com/NixOS/nixpkgs/commit/03e83a08933b76dbe742c91750ee5b1f759ecda0, needs partial backport.

GNOME Shell is primarily used for installing Shell extensions, optionally it can update some firmwares when fwupd is enabled. In the future, we should get [Nix backend](https://github.com/NixOS/nixpkgs/issues/21230) and also [Flatpak](https://github.com/NixOS/nixpkgs/pull/33371) support.

**Comment by davidak** [comment]
That are some great news!

**Comment by matthewbauer** [comment]
One thing on the Nix backend though: it will require Appstream data to work IIRC (#15932). There is another app that just handles PackageKit stuff but I can’t remember the name.

**Comment by jtojnar** [comment]
@matthewbauer [discover](https://github.com/NixOS/nixpkgs/pull/34756)?

**Comment by davidak** [comment]
@matthewbauer jou probably don't mean [AppCenter](https://github.com/elementary/appcenter) for elementary OS.

**Comment by davidak** [comment]
The problem is fixed in 18.09.

I'm able to install GNOME Shell extensions.

![screenshot from 2018-11-19 16-07-53](https://user-images.githubusercontent.com/91113/48718097-0364a600-ec1b-11e8-89de-9263a5dca29a.png)

When Flatpack support is enabled and Flathub configured i even can install Flatpack packages. Perfect!

![screenshot from 2018-11-19 16-41-08](https://user-images.githubusercontent.com/91113/48718167-21caa180-ec1b-11e8-9e69-8bd91bd579d7.png)

![screenshot from 2018-11-19 16-44-57](https://user-images.githubusercontent.com/91113/48718171-255e2880-ec1b-11e8-985b-a96be388daac.png)
